### PR TITLE
Switch to auto-formatted code with Black

### PR DESCRIPTION
**Pull request by bkgoksel**
Black(https://github.com/ambv/black) is a more opinionated code formatter than YAPF. It's less customizable than YAPF but seems to produce better results by default. It also checks against functionality regressions by constructing ASTs for code before and after formatting and asserting nothing has changed.

It requires Python 3.6 to run and expects code compliant with Python3 syntax, so this PR also includes changes that make our codebase syntax-wise compliant with Python 3 (mainly print calls with parentheses, `except Exception as e` instead of `except Exception, e` and converting octets from `0770` to `0o770`)

### CODE REVIEW

File: alembic/versions/2014080613_add_owner_id_to_bundle_341ee10697f1.py [hunk 1]
```diff
@@ -7,7 +7,7 @@
 """
 
 # revision identifiers, used by Alembic.
-revision = '341ee10697f1'
+revision = "341ee10697f1"
```
Comment (percyliang) [inline alembic/versions/2014080613_add_owner_id_to_bundle_341ee10697f1.py]: Can we just stick with single quotes throughout?

File: alembic/versions/2014090815_switching_owner_id_to_str_143201389156.py [hunk 2]
```diff
@@ -7,24 +7,60 @@
...
 from alembic import op
 import sqlalchemy as sa
 
 
 def upgrade():
     ### commands auto generated by Alembic - please adjust! ###
-    op.alter_column("bundle", 'owner_id', type_=sa.String(255), existing_type=sa.Integer(), nullable=True)
-    op.alter_column("worksheet", 'owner_id', type_=sa.String(255), existing_type=sa.Integer(), nullable=True)
-    op.alter_column("group", 'owner_id', type_=sa.String(255), existing_type=sa.Integer(), nullable=True)
+    op.alter_column(
```
Comment (percyliang) [inline alembic/versions/2014090815_switching_owner_id_to_str_143201389156.py]: Make it not expand?

**Comment by bkgoksel** [comment]
Started from scratch, switched off string quotation change.

My opinion is that having auto formatting by default and manually changing formatting around the code base as we run into non-optimal things is a better approach than the opposite.

**Comment by percyliang** [comment]
I agree with that generally.

How do we make sure the exceptional changes don't get clobbered by the
automatic formatting?

We should tweak things so that the automatic changes are as good as
possible to minimize the gap.

**Comment by bkgoksel** [comment]
Any code wrapped in a pair of `# fmt: off` `# fmt: on` comments doesn't get modified or flagged as not compliant with formatting. This is mainly useful for things like argument lists where we would tolerate going over the line length limit for compactness/uniformity.

I was talking to a friend who is trying to set up auto formatting for a large python code base where he works and he also claimed that Black was the best one they've tried, but unfortunately it doesn't offer any customization beyond string quote standardization (on or off, off for us) and line length limit (default 88, 100 for us)

File: codalab/bundles/derived_bundle.py [hunk 3]
```diff
@@ -10,31 +10,41 @@
 
 class DerivedBundle(NamedBundle):
     METADATA_SPECS = list(NamedBundle.METADATA_SPECS)
-    METADATA_SPECS.append(MetadataSpec('allow_failed_dependencies', bool, 'Whether to allow this bundle to have failed or killed dependencies.', default=False))
+    # Don't format specs
+    # fmt: off
+    METADATA_SPECS.append(
```
Comment (percyliang) [inline codalab/bundles/derived_bundle.py:15]: Thought turning formatting off would leave it one line.

File: codalab/client/json_api_client.py [hunk 4]
```diff
@@ -1,14 +1,11 @@
 import httplib
 import socket
 import sys
+import six
```
Comment (percyliang) [inline codalab/client/json_api_client.py:4]: This seems to be semantics changing...

File: worker/codalabworker/local_run/local_run_manager.py [hunk 5]
```diff
@@ -20,15 +20,26 @@ class LocalRunManager(BaseRunManager):
...
     # Network buffer size to use while proxying with netcat
     NETCAT_BUFFER_SIZE = 4096
     # Number of seconds to wait for bundle kills to propagate before forcing kill
     KILL_TIMEOUT = 100
     # Directory name to store running bundles in worker filesystem
     BUNDLES_DIR_NAME = 'runs'
 
-    def __init__(self, worker, docker, image_manager, dependency_manager,
-                 commit_file, cpuset, gpuset, work_dir, docker_network_prefix='codalab_worker_network'):
+    def __init__(
```
Comment (percyliang) [inline worker/codalabworker/local_run/local_run_manager.py:31]: Is there a way to prevent this from happening?

**Review by percyliang — APPROVED** [review]
Generally looks good.  Don't like everything, but I think it's net positive.

File: codalab/bundles/derived_bundle.py (hunk 3 above)
Comment (bkgoksel) [reply codalab/bundles/derived_bundle.py:15]: I just hand modified those back, forgot to put the append and the actual spec on the same line, can do that.

File: codalab/client/json_api_client.py (hunk 4 above)
Comment (bkgoksel) [reply codalab/client/json_api_client.py:4]: This was a manual change I made to make our code base syntax compatible with both Python 2 and 3 as per https://pythonhosted.org/six/index.html#six.reraise
 Python 3 syntax compatibility is necessary for this formatter as it builds ASTs and compares them to ensure functionality stays the same but does so in Python 3.6

File: worker/codalabworker/local_run/local_run_manager.py (hunk 5 above)
Comment (bkgoksel) [reply worker/codalabworker/local_run/local_run_manager.py:31]: This library has a strong tendency to start argument lists on new lines and separating each argument into its own line. I read that their rationale behind this style is to have minimal/clearer diffs when arguments change.
I'm personally neutral about this approach but if it's not something we want to adopt then we should probably reconsider using Black as it is really unavoidable and will probably happen anywhere many arguments are passed.

Comment (percyliang) [reply worker/codalabworker/local_run/local_run_manager.py:31]: I think that's logical.

**Pull request merged**